main/adapters/repository/database.go

```diff
@@ -245,19 +303,39 @@ func (dm *DatabaseManager) retry(f func() error) (err error) {
 }
 
 func (dm *DatabaseManager) isRecoverable(err error) bool {
-	if pqErr, ok := err.(*pq.Error); ok {
-		switch pqErr.Code {
-		case "42P01": // undefined_table
-			err = dm.CreateTable(PostgresIdentity)
-			if err != nil {
-				log.Errorf("creating DB table failed: %v", err)
+	switch dm.driverName {
+	case PostgreSQL:
+		if pqErr, ok := err.(*pq.Error); ok {
+			switch pqErr.Code {
+			case "55P03", "53300", "53400": // lock_not_available, too_many_connections, configuration_limit_exceeded
+				time.Sleep(10 * time.Millisecond)
+				return true
+			case "42P01": // undefined_table
+				err = dm.CreateTable(PostgresIdentity)
+				if err != nil {
+					log.Errorf("creating DB table failed: %v", err)
+				}
+				return true
+			}
+			log.Errorf("unexpected postgres database error: %s", pqErr)
+		}
+	case SQLite:
+		if liteErr, ok := err.(*sqlite.Error); ok {
+			if liteErr.Code() == 5 || // SQLITE_BUSY
+				liteErr.Code() == 6 || // SQLITE_LOCKED
+				liteErr.Code() == 261 { // SQLITE_BUSY_RECOVERY
+				time.Sleep(10 * time.Millisecond)
+				return true
+			}
+			if liteErr.Code() == 1 { // SQLITE_ERROR
+				err = dm.CreateTable(SQLiteIdentity)
+				if err != nil {
+					log.Errorf("creating DB table failed: %v", err)
+				}
+				return true
 			}
-			return true
-		case "55P03", "53300", "53400": // lock_not_available, too_many_connections, configuration_limit_exceeded
-			time.Sleep(10 * time.Millisecond)
-			return true
+			log.Errorf("unexpected sqlite database error: %s", liteErr)
 		}
-		log.Errorf("%s = %s", err, pqErr.Code)
 	}
 	return false
```

Comment:
In case the database can not be reached because there are too many connections being used already or there is an ongoing transaction, we wait for a few milliseconds before we try to connect again. Do you think it would be better to not wait?

Reply:
Ah ok, i see. No, i think it's fine how it is done!